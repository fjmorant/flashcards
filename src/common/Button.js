import React from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  StyleSheet,
} from 'react-native';
import theme from './defaultTheme';
import {PRIMARY_COLOR, DISABLED_PRIMARY_COLOR} from './ThemeConstants';

const styles = StyleSheet.create({
  container: {
    borderRadius: 5,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default ({
  disabled,
  title,
  height,
  onPress,
}) => (
  <TouchableOpacity
      onPress={!disabled ? onPress : null}
      activeOpacity={disabled ? 1 : 0}>
    <View style={[styles.container, {height, backgroundColor: !disabled ? theme[PRIMARY_COLOR] : theme[DISABLED_PRIMARY_COLOR]}]}>
      <Text style={{color: 'white'}}>{title}</Text>
    </View>
  </TouchableOpacity>
);
