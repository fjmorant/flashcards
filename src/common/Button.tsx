import * as React from 'react'
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native'
import theme from './defaultTheme'
import {DISABLED_PRIMARY_COLOR, PRIMARY_COLOR} from './ThemeConstants'

const styles = StyleSheet.create({
  container: {
    borderRadius: 5,
    justifyContent: 'center',
    alignItems: 'center',
  },
})

export interface IProps {
  disabled: boolean
  title: string
  titleSize?: number
  height: number
  onPress?: () => void
}

export default class Button extends React.Component<IProps> {
  public render() {
    const {disabled, title, titleSize = 20, height, onPress} = this.props

    return (
      <TouchableOpacity
        activeOpacity={disabled ? 1 : 0}
        onPress={!disabled ? onPress : () => null}>
        <View
          style={[
            styles.container,
            {
              height,
              backgroundColor: !disabled
                ? theme[PRIMARY_COLOR]
                : theme[DISABLED_PRIMARY_COLOR],
            },
          ]}>
          <Text style={{color: 'rgb(255, 255, 255)', fontSize: titleSize}}>
            {title}
          </Text>
        </View>
      </TouchableOpacity>
    )
  }
}
