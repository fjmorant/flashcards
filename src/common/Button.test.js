import 'react-native'
import React from 'react'
import Button from './Button'

import renderer from 'react-test-renderer'

describe('Button', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Button onPress={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders without onPress', () => {
    const tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('disabled', () => {
    const tree = renderer.create(<Button disabled />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
