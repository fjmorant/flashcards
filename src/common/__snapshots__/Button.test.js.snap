// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button disabled 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(0, 64, 128, 0.5)",
          "height": undefined,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgb(255, 255, 255)",
          "fontSize": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgb(0, 64, 128)",
          "height": undefined,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`Button renders without onPress 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgb(0, 64, 128)",
          "height": undefined,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgb(0, 64, 128)",
          "height": undefined,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }
    />
  </View>
</View>
`;
